src/pre_process/m_compute_levelset.fpp

```diff
@@ -315,23 +316,25 @@ contains
                         if (side_dists(2) == 0) then
                             levelset_norm%sf(i, j, 0, ib_patch_id, 1) = 0d0
                         else
-                            levelset_norm%sf(i, j, 0, ib_patch_id, 1) = side_dists(2)/ &
+                            levelset_norm%sf(i, j, 0, ib_patch_id, 1) = -side_dists(2)/ &
```

Comment:
This subroutine has several minor changes, and I can't tell what their purpose is. I see that the simulation results for a 2D rectangle are correct, so I'm guessing these are a result of adding the cuboid patch?

Reply:
That was a bug and @anshgupta1234 can confirm this.

Reply:
it seems like @wilfonba is saying there were several changes in this subroutine. Were they all bugs?

Reply:
> it seems like @wilfonba is saying there were several changes in this subroutine. Were they all bugs?

Yes

Reply:
Yep, this subroutine had some bugs we had found earlier, which is why the analytical & stl patches weren't returning the same results. These changes fix it